From 6a6a442585fcdf534cc448706a1be2d460862bdf Mon Sep 17 00:00:00 2001
From: Viktor Tusa <tusa@balabit.hu>
Date: Wed, 26 Mar 2014 17:05:08 +0100
Subject: [PATCH] value-pairs: Fixed a memory leak in value-pairs.

We should free the elements of the GPtrArray too, not just the GPtrArray.

Signed-off-by: Viktor Tusa <tusa@balabit.hu>

--- lib/value-pairs.c.orig	2014-03-27 18:16:45.492827781 -0400
+++ lib/value-pairs.c	2014-03-27 18:16:56.785426734 -0400
@@ -651,6 +651,7 @@
      NULL), so treat that normally. */
   key = g_strdup(g_ptr_array_index(tokens, tokens->len - 1));
 
+  g_ptr_array_foreach(tokens, g_free, NULL);
   g_ptr_array_free(tokens, TRUE);
 
   return key;
